File: src-tauri/src/commands/cron.rs
use crate::models::result::CommandResult;
use crate::services::cron_service::{
    self, CreateCronJobPayload, CronJob, DeleteCronJobData, TriggerCronJobData,
    UpdateCronJobPayload,
};

#[tauri::command]
pub async fn list_cron_jobs() -> CommandResult<Vec<CronJob>> {
    match cron_service::list_cron_jobs().await {
        Ok(data) => CommandResult::ok(data),
        Err(error) => CommandResult::err(error),
    }
}

#[tauri::command]
pub async fn create_cron_job(payload: CreateCronJobPayload) -> CommandResult<CronJob> {
    match cron_service::create_cron_job(payload).await {
        Ok(data) => CommandResult::ok(data),
        Err(error) => CommandResult::err(error),
    }
}

#[tauri::command]
pub async fn update_cron_job(payload: UpdateCronJobPayload) -> CommandResult<CronJob> {
    match cron_service::update_cron_job(payload).await {
        Ok(data) => CommandResult::ok(data),
        Err(error) => CommandResult::err(error),
    }
}

#[tauri::command]
pub async fn delete_cron_job(id: String) -> CommandResult<DeleteCronJobData> {
    match cron_service::delete_cron_job(id).await {
        Ok(data) => CommandResult::ok(data),
        Err(error) => CommandResult::err(error),
    }
}

#[tauri::command]
pub async fn trigger_cron_job(id: String) -> CommandResult<TriggerCronJobData> {
    match cron_service::trigger_cron_job(id).await {
        Ok(data) => CommandResult::ok(data),
        Err(error) => CommandResult::err(error),
    }
}

File: src/main/ipc/cron.ts
import { ipcMain } from "electron";
import { IPC } from "../../shared/ipc-channels.js";
import { ok, err } from "../models/result.js";
import { toAppError } from "./utils.js";
import {
  listCronJobs,
  createCronJob,
  updateCronJob,
  deleteCronJob,
  triggerCronJob,
} from "../services/cron-service.js";
import type {
  CreateCronJobPayload,
  UpdateCronJobPayload,
} from "../services/cron-service.js";

export function registerCronHandlers(): void {
  ipcMain.handle(IPC.LIST_CRON_JOBS, async () => {
    try { return ok(await listCronJobs()); }
    catch (error) { return err(toAppError(error)); }
  });

  ipcMain.handle(IPC.CREATE_CRON_JOB, async (_event, payload: CreateCronJobPayload) => {
    try { return ok(await createCronJob(payload)); }
    catch (error) { return err(toAppError(error)); }
  });

  ipcMain.handle(IPC.UPDATE_CRON_JOB, async (_event, payload: UpdateCronJobPayload) => {
    try { return ok(await updateCronJob(payload)); }
    catch (error) { return err(toAppError(error)); }
  });

  ipcMain.handle(IPC.DELETE_CRON_JOB, async (_event, payload: { id: string }) => {
    try { return ok(await deleteCronJob(payload.id)); }
    catch (error) { return err(toAppError(error)); }
  });

  ipcMain.handle(IPC.TRIGGER_CRON_JOB, async (_event, payload: { id: string }) => {
    try { return ok(await triggerCronJob(payload.id)); }
    catch (error) { return err(toAppError(error)); }
  });
}
